sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py

```diff
@@ -129,33 +129,24 @@ class DirectoryProperties(DictMixin):
     :var ~azure.storage.filedatalake.ContentSettings content_settings:
     """
     def __init__(self, **kwargs):
-        super(DirectoryProperties, self).__init__(
-            **kwargs
-        )
-        self.name = None
-        self.etag = None
+        self.name = kwargs.get('name')
+        self.etag = kwargs.get('ETag')
         self.deleted = None
-        self.metadata = None
-        self.lease = None
-        self.last_modified = None
-        self.creation_time = None
+        self.metadata = kwargs.get('metadata')
```

Comment:
should these all be `kwargs.get('metadata', None)`

Reply:
That's not needed - `get` already returns None by default.